client/pk-monitor.c

```diff
@@ -128,6 +128,42 @@ pk_monitor_adopt_cb (PkClient *_client, GAsyncResult *res, gpointer user_data)
 	}
 }
 
+static gchar*
+pk_monitor_get_caller_info (GDBusProxy *bus_proxy, const gchar *name)
+{
+	gboolean ret;
+	gchar *cmdline = NULL;
+	g_autofree gchar *filename = NULL;
+	g_autoptr(GError) error = NULL;
+	guint pid = G_MAXUINT;
+	g_autoptr(GVariant) value = NULL;
+
+	/* get pid from DBus */
+	value = g_dbus_proxy_call_sync (bus_proxy,
+					"GetConnectionUnixProcessID",
+					g_variant_new ("(s)",
+						       name),
+					G_DBUS_CALL_FLAGS_NONE,
+					2000,
+					NULL,
+					&error);
+	if (value == NULL) {
+		g_warning ("Failed to get pid for %s: %s",
+			   name, error->message);
+		return g_strdup_printf ("bus:%s", name);;
+	}
+	g_variant_get (value, "(u)", &pid);
+
+	/* get command line from proc */
+	filename = g_strdup_printf ("/proc/%i/cmdline", pid);
+	ret = g_file_get_contents (filename, &cmdline, NULL, &error);
+	if (!ret) {
+		/* we failed to get the command-line, maybe we don't have permission */
+		return g_strdup_printf ("pid:%i", pid);
+	}
+	return cmdline;
```

Comment:
Returning `/proc/*/cmdline` directly will give a string with nul bytes between the arguments. i.e. It’ll appear to only be the `argv[0]`? Is that what you want? If so, it might clarify things to add a comment about that decision.

Otherwise, you’ll have to post-process the `cmdline` to replace `\0` with ` ` up to the file length.